WcaOnRails/app/webpacker/lib/utils/wcif.js

```diff
@@ -86,6 +86,24 @@ export function venueWcifFromRoomId(scheduleWcif, id) {
   return scheduleWcif.venues.find((venue) => venue.rooms.some((room) => room.id === intId));
 }
 
+export function activityWcifFromId(scheduleWcif, id) {
+  return scheduleWcif.venues.flatMap(
+    ({ rooms }) => rooms.flatMap(({ activities }) => activities),
+  ).find((activity) => activity.id === id);
+}
+
+export function doActivitiesMatch(a1, a2) {
+  return a1.activityCode === a2.activityCode
+    && a1.startTime === a2.startTime
+    && a1.endTime === a2.endTime;
+}
+
+export function getMatchingActivities(scheduleWcif, activity) {
+  return scheduleWcif.venues.flatMap(
+    ({ rooms }) => rooms.flatMap(({ activities }) => activities),
+  ).filter((act) => doActivitiesMatch(act, activity));
```

Comment:
65% sure we already have a helper method that "flattens out" all activities from a venue (might be my TNoodle brain though and the helper method exists in the codebase over there). Please double-check to make sure :)

Reply:
I don't see one in `lib/utils` anywhere.

Reply:
Okay, then it must have been TNoodle :smile: